fuse_examples/classification/MG_CMMD/dataset.py

```diff
@@ -0,0 +1,215 @@
+                                       gt_processors=gt_processors,
+                                       post_processing_func=post_cache_processing_func,
+                                       augmentor=augmentor,
+                                       visualizer=visualiser)
+
+    lgr.info(f'- Load and cache data:')
+    train_dataset.create(reset_cache=reset_cache)
+    lgr.info(f'- Load and cache data: Done')
+
+    # Create validation data source
+    validation_data_source = FuseDataSourceFolds(input_source=input_source_gt,
+                                            input_df=None,
+                                            phase='validation',
+                                            no_mixture_id='ID1',
+                                            balance_keys=[target],
+                                            reset_partition_file=False,
+                                            folds=[3],
+                                            num_folds=5,
+                                            partition_file_name=partition_file_path)
+
+    ## Create dataset
+    validation_dataset = FuseDatasetDefault(cache_dest=cache_dir,
+                                            data_source=validation_data_source,
+                                            input_processors=input_processors,
+                                            gt_processors=gt_processors,
+                                            post_processing_func=post_cache_processing_func,
+                                            augmentor=None,
+                                            visualizer=visualiser)
+
+    test_data_source = FuseDataSourceFolds(input_source_gt,None,'test','ID1',[target],False,[4],5,partition_file_path)
+
+    test_dataset_source =  FuseDataSourceFolds(input_source=input_source_gt,
+                                            input_df=None,
+                                            phase='test',
+                                            no_mixture_id='ID1',
+                                            balance_keys=[target],
+                                            reset_partition_file=False,
+                                            folds=[4],
+                                            num_folds=5,
+                                            partition_file_name=partition_file_path)
+    test_dataset = FuseDatasetDefault(cache_dest=cache_dir,
+                                            data_source=test_dataset_source,
+                                            input_processors=input_processors,
+                                            gt_processors=gt_processors,
+                                            post_processing_func=post_cache_processing_func,
+                                            augmentor=None,
+                                            visualizer=visualiser)
+
+    lgr.info(f'- Load and cache data:')
+    validation_dataset.create(
```

Comment:
This should to be right after you create the validation dataset.
And here you missed test_dataset.create(..)

Reply:
ok, fixed